File: rb_101/exam_prep/reverse.rb
# input: a string
# output: the string, reversed

# set i = 0
# set j = string.length - 1
# while i < string.length / 2
#   switch the characters at i and j
#   increment i and decrement j

def reverse(string)
  i = 0
  j = string.length - 1
  while i < string.length / 2
    string[i], string[j] = string[j], string[i]
    i += 1
    j -= 1
  end
  string
end

p reverse('hello world')
p reverse('three')

File: js_210/fundamentals/arrays/reverse.js
function reverse(inputForReversal) {
  let output;
  if (Array.isArray(inputForReversal)) {
    output = [];
  } else {
    output = '';
  }

  for (let i = inputForReversal.length - 1; i >= 0; i--) {
    if (Array.isArray(output)) {
      output.push(inputForReversal[i]);
    } else {
      output += inputForReversal[i];
    }
  }
  return output;
}

console.log(reverse('Hello'));
console.log(reverse('a'));
console.log(reverse([1, 2, 3, 4]));
console.log(reverse([]));

const array = [1, 2, 3];
reverse(array);
console.log(array);